### ISSUE TITLE
Expired Key For 'fs-uae' Via Ubuntu 22.04 Instructions

### ISSUE THREAD

**Issue by SamuelBanya** [FrodeSolheim/fs-uae#323]
tldr; You need to fix the expired GPG key for 'fs-uae' on the related OpenSuse repos.

I followed the following steps as per the Ubuntu 22.04 instructions since I'm running Linux Mint on the older computer I'm using for retro stuff:
`
echo 'deb http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/xUbuntu_22.04/ /' | sudo tee /etc/apt/sources.list.d/home:FrodeSolheim:stable.list
curl -fsSL https://download.opensuse.org/repositories/home:FrodeSolheim:stable/xUbuntu_22.04/Release.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/home_FrodeSolheim_stable.gpg > /dev/null
sudo apt update
sudo apt install fs-uae fs-uae-launcher fs-uae-arcade
`

I get the following output:
`
sam@retrostation:/etc/apt/trusted.gpg.d$ sudo apt update
Ign:1 http://packages.linuxmint.com vera InRelease
Hit:2 http://packages.linuxmint.com vera Release                               
Hit:4 http://archive.ubuntu.com/ubuntu jammy InRelease                         
Get:5 http://security.ubuntu.com/ubuntu jammy-security InRelease [110 kB]
Get:6 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]        
Get:7 http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/xUbuntu_22.04  InRelease [1,557 B]
Err:7 http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/xUbuntu_22.04  InRelease
  The following signatures were invalid: EXPKEYSIG BF2C3F2504BFEF00 home:FrodeSolheim OBS Project <home:FrodeSolheim@build.opensuse.org>
Get:8 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [107 kB]
Reading package lists... Done     
W: GPG error: http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/xUbuntu_22.04  InRelease: The following signatures were invalid: EXPKEYSIG BF2C3F2504BFEF00 home:FrodeSolheim OBS Project <home:FrodeSolheim@build.opensuse.org>
E: The repository 'http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/xUbuntu_22.04  InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
`

I also confirmed the presence of the GPG key with this output as well:
`
sam@retrostation:/etc/apt/trusted.gpg.d$ file /etc/apt/trusted.gpg.d/home_FrodeSolheim_stable.gpg
/etc/apt/trusted.gpg.d/home_FrodeSolheim_stable.gpg: OpenPGP Public Key Version 4, Created Sun Nov 29 18:20:09 2020, RSA (Encrypt or Sign, 2048 bits); User ID; Signature; OpenPGP Certificate

`

**Comment by Recydywa** [comment]
I also get the key expired error on Ubuntu 22.04 LTS.

And this started a week or two weeks ago.

**Comment by maidentaiwan** [comment]
This has been going on for a few weeks now. Any plans to fix it?

**Comment by tikikevin** [comment]
I still can't install FS-UAE under Ubuntu 22.04 LTS. Please help. Is there anyone to contact?

**Comment by HAL9081** [comment]
Same here. Can you please update the key. We are not able to use fs-uae-launcher on Ubuntu22.04.

gpg --verify InRelease
gpg: Signature made Mo 02 Mai 2022 16:06:48 CEST
gpg:                using RSA key BF2C3F2504BFEF00
gpg: Good signature from "home:FrodeSolheim OBS Project <home:FrodeSolheim@build.opensuse.org>" [expired]
gpg: Note: This key has expired!
Primary key fingerprint: BE6D C98E 720C C00E 404F  129C BF2C 3F25 04BF EF00

Thank you very much!

**Comment by maidentaiwan** [comment]
I have tried emailing the author (Frode) without any response, and I tried signing up on his support board (http://eab.abime.net/forumdisplay.php?f=122) but the signup is still pending. I wonder if something has happened to Frode.

**Comment by FrodeSolheim** [comment]
Hi, apologies for not responding earlier! I'm OK!

I think the package signature keys have simply timed out, and for some reason, the OBS service does not issue new ones / resign the repository automatically. I'm trying a workaround now by pushing a dummy update to the fs-uae package which will trigger a rebuild and re-publish of the repository. That might be enough to sort it out.

**Comment by maidentaiwan** [comment]
Whew, I am glad you are OK. Thanks so much, Frode!

**Comment by Recydywa** [comment]
Thank you it worked.

I removed the key added it back again and the refresh of apt was successful. No more expired key complaints.

**Comment by maidentaiwan** [comment]
Worked for me too on Ubuntu 22.04.

```
$ sudo rm /etc/apt/trusted.gpg.d/home_FrodeSolheim_stable.gpg
$ curl -fsSL https://download.opensuse.org/repositories/home:FrodeSolheim:stable/xUbuntu_22.04/Release.key | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/home_FrodeSolheim_stable.gpg > /dev/null
$ sudo apt update
```

**Comment by olispeedy** [comment]
the keyring seems to be outdated or so again .. at least for debian testing (trixie) I get while calling 'apt update' :

```
Err:1 http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_Testing  InRelease                                                                    
  Sub-process /usr/bin/sqv returned an error code (1), error message is: Missing key BF2C3F2504BFEF00, which is needed to verify signature.
Warning: GPG error: http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_Testing  InRelease: Sub-process /usr/bin/sqv returned an error code (1), error message is: Missing key BF2C3F2504BFEF00, which is needed to verify signature.
Error: The repository 'http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_Testing  InRelease' is not signed.
Notice: Updating from such a repository can't be done securely, and is therefore disabled by default.
```

also the "workaround" seems failing - my source :
`deb [trusted=yes,allow-insecure=yes,signed-by=/etc/apt/keyrings/FrodeSolheim_stable.gpg] http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_Testing/ /`

I've put the "/etc/apt/keyrings/FrodeSolheim_stable.gpg" as follows :
`curl -fsSL http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_Testing/Release.gpg | gpg --dearmor | sudo tee /etc/apt/keyrings/FrodeSolheim_stable.gpg > /dev/null`

**Comment by FrodeSolheim** [comment]
Did another push to the repo(s), also added support for more distros. Closing (please reopen if this did not fix the key signing issue).

**Comment by olispeedy** [comment]
still same issue with the updated signature / keyring :

```
Err:5 http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_12  InRelease             
  Sub-process /usr/bin/sqv returned an error code (1), error message is: Missing key BF2C3F2504BFEF00, which is needed to verify signature.
Hit:6 http://www.deb-multimedia.org trixie InRelease            
Warning: GPG error: http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_12  InRelease: Sub-process /usr/bin/sqv returned an error code (1), error message is: Missing key BF2C3F2504BFEF00, which is needed to verify signature.
Error: The repository 'http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_12  InRelease' is not signed.
Notice: Updating from such a repository can't be done securely, and is therefore disabled by default.
Notice: See apt-secure(8) manpage for repository creation and user configuration details.
```

**Comment by FrodeSolheim** [comment]
Did you also try adding the new (?) key:

```
echo 'deb http://download.opensuse.org/repositories/home:/FrodeSolheim:/stable/Debian_12/ /' | sudo tee /etc/apt/sources.list.d/home:FrodeSolheim:stable.list
curl -fsSL https://download.opensuse.org/repositories/home:FrodeSolheim:stable/Debian_12/Release.key | gpg --dearmor | sudo tee
/etc/apt/trusted.gpg.d/home_FrodeSolheim_stable.gpg > /dev/null
sudo apt update
sudo apt install fs-uae fs-uae-launcher fs-uae-arcade
```

**Comment by olispeedy** [comment]
yes I did .. looks a bit like the signing method is a bit outdated ... other 3rdParty repos that e.g. use [aptly](https://www.aptly.info/) work
